src/web/lib/storage.js

```diff
@@ -17,16 +17,32 @@ const generateThemeKey = () =>
   `${Date.now()}-${Math.floor(Math.random() * 1000)}`;
 
 function putTheme(key, theme) {
-  log("putTheme", key, theme);
-  const storageKey = themeStorageKey(key);
-  localStorage.setItem(
-    storageKey,
-    JSON.stringify({
-      theme,
-      modified: Date.now()
-    })
-  );
-  notifySelfForStorage(storageKey);
+  checkDuplicateTheme(theme, (isThemeDuplicate) => {
+    if (!isThemeDuplicate) {
+      log("putTheme", key, theme);
+      const storageKey = themeStorageKey(key);
+      localStorage.setItem(
+        storageKey,
+        JSON.stringify({
+          theme,
+          modified: Date.now()
+        })
+      );
+      notifySelfForStorage(storageKey);
+    }
+  });
+}
+
+function checkDuplicateTheme(theme, callback) {
+  const themesList = listThemes();
+  let isThemeDuplicate = false;
+  for (const key of Object.keys(themesList)) {
+    isThemeDuplicate = themesEqual(theme, themesList[key].theme);
+    if (isThemeDuplicate) {
+      break;
+    }
+  }
```

Comment:
This could be written a little more cleanly, maybe something like:

``` js
return Object.keys(listThemes()).filter((key) => {
  return themesEqual(theme, themesList[key].theme);
}).length > 0;
```

Reply:
@meandavejustice: Wouldn't we want something like [`Array#some()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some) [instead of `Array#filter()`] which should exit the loop as soon as a match is found?

Reply:
@pdehaan Sure thing, `some` is even better!

Reply:
Thanks for this suggestion too. I learnt a new hack.